x-pack/plugins/ml/public/application/components/model_snapshots/model_snapshots_table.tsx

```diff
@@ -0,0 +1,318 @@
+      )}
+
+      {closeJobModalVisible !== null && combinedJobState !== null && (
+        <EuiOverlayMask>
+          <EuiConfirmModal
+            title={
+              combinedJobState === COMBINED_JOB_STATE.OPEN_AND_RUNNING
+                ? i18n.translate('xpack.ml.modelSnapshotTable.closeJobConfirm.stopAndClose.title', {
+                    defaultMessage: 'Stop datafeed and close job?',
+                  })
+                : i18n.translate('xpack.ml.modelSnapshotTable.closeJobConfirm.close.title', {
+                    defaultMessage: 'Close job?',
+                  })
+            }
+            onCancel={hideCloseJobModalVisible}
+            onConfirm={forceCloseJob}
+            cancelButtonText={i18n.translate(
+              'xpack.ml.modelSnapshotTable.closeJobConfirm.cancelButton',
+              {
+                defaultMessage: 'Cancel',
+              }
+            )}
+            confirmButtonText={
+              combinedJobState === COMBINED_JOB_STATE.OPEN_AND_RUNNING
+                ? i18n.translate(
+                    'xpack.ml.modelSnapshotTable.closeJobConfirm.stopAndClose.button',
+                    {
+                      defaultMessage: 'Force stop and close',
+                    }
+                  )
+                : i18n.translate('xpack.ml.modelSnapshotTable.closeJobConfirm.close.button', {
+                    defaultMessage: 'Force close',
+                  })
+            }
+            defaultFocusedButton="confirm"
+          >
+            <p>
+              {combinedJobState === COMBINED_JOB_STATE.OPEN_AND_RUNNING && (
+                <FormattedMessage
+                  id="xpack.ml.modelSnapshotTable.closeJobConfirm.contentOpenAndRunning"
+                  defaultMessage="Job is currently open and running."
+                />
+              )}
+              {combinedJobState === COMBINED_JOB_STATE.OPEN_AND_STOPPED && (
+                <FormattedMessage
+                  id="xpack.ml.modelSnapshotTable.closeJobConfirm.contentOpen"
+                  defaultMessage="Job is currently open."
+                />
+              )}
+              <br />
+              <FormattedMessage
+                id="xpack.ml.modelSnapshotTable.closeJobConfirm.content"
+                defaultMessage="Snapshot revert can only happen on jobs which are closed."
+              />
+            </p>
+          </EuiConfirmModal>
+        </EuiOverlayMask>
+      )}
+    </>
+  );
+};
+
+function renderDate(date: number) {
+  return formatDate(date, TIME_FORMAT);
+}
+
+async function getCombinedJobState(jobId: string) {
+  const jobs = await ml.jobs.jobs([jobId]);
+
+  if (jobs.length !== 1) {
+    return COMBINED_JOB_STATE.UNKNOWN;
+  }
+
+  if (jobs[0].state !== JOB_STATE.CLOSED) {
+    if (jobs[0].datafeed_config.state !== DATAFEED_STATE.STOPPED) {
+      return COMBINED_JOB_STATE.OPEN_AND_RUNNING;
+    }
+    return COMBINED_JOB_STATE.OPEN_AND_STOPPED;
+  }
+  return COMBINED_JOB_STATE.CLOSED;
+}
```

Comment:
This logic should reside inside of the Jobs service.

Reply:
This state is only used in this file because it is reliant on knowing the combination of the two states.
the combined job concept itself is a workaround because of our historical implementation of having the datafeed inside the job.
so I'm reluctant to move it to a common location because it's not an "official" job or datafeed state and may reinforce this idea of a combined job.